api/pkg/service/catalog/syncer.go

```diff
@@ -404,3 +409,56 @@ func (s *syncer) updateCatalogResult(
 	}
 	return nil
 }
+
+// Delete the resource by id
+func (s *syncer) deleteResource(txn *gorm.DB, log *zap.SugaredLogger, res model.Resource) {
+
+	txn.Unscoped().Where("id = ?", res.ID).Delete(&model.Resource{})
```

Comment:
on what condition are we deleting resource?

Reply:
deleting all and adding again?

Reply:
it compares db resources and sync resources by `Kind and Name` and find all resources which are in db and not in sync resources list and then deletes one by one resource

Reply:
ah okay

Reply:
what if we add all resources in db as current flow and save the id of resources in array and then use that to delete resources?
delete resources whose id is not in the array

Reply:
similar to your approach
wdyt

Reply:
yes,it would be a better approach than current